scripts/luax.lua

```diff
@@ -110,6 +189,35 @@ function table.filter(list, predicate)
     end
 end
 
+--- Remove all items from the list.
+--
+-- Use this function in particular if you have several references to the list.
+--
+-- As usual, non-list keys are ignored.
+-- This means, `#list` is 0 after `clear`, but there might still be other keys.
+--
+-- @tparam table list the list to clear
+-- @usage
+-- -- do
+-- local list = {"a", "b"}
+-- local ref = list
+-- print(#list, #ref) -- 2   2
+-- table.clear(list)
+-- print(#list, #ref) -- 0   0
+-- print(list == ref) -- true
+-- -- (usually) do not
+-- local list = {"a", "b"}
+-- local ref = list
+-- print(#list, #ref) -- 2   2
+-- list = {}
+-- print(#list, #ref) -- 0   2
+-- print(list == ref) -- false
+function table.clear(list)
```

Comment:
Itn't `list = {}` much more efficient?

Reply:
Of course, but it creates a new table, which might not be what you need. See the example.